src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -117,86 +118,116 @@ public bool TryRemoveContext(IAzureContext context)
                  !account.IsPropertySet(AzureAccount.Property.CertificateThumbprint))
                 ? ShowDialog.Always : ShowDialog.Never;
 
-            // (tenant and subscription are present) OR
-            // (tenant is present and subscription is not provided)
-            if (!string.IsNullOrEmpty(tenantId))
+            if (skipValidation)
             {
-                var token = AcquireAccessToken(
-                    account, 
-                    environment, 
-                    tenantId, 
-                    password, 
-                    promptBehavior,
-                    promptAction);
-                if (TryGetTenantSubscription(
-                    token, 
-                    account, 
-                    environment, 
-                    tenantId, 
-                    subscriptionId, 
-                    subscriptionName, 
-                    out newSubscription, 
-                    out newTenant))
+                if ((string.IsNullOrEmpty(subscriptionId) && string.IsNullOrEmpty(subscriptionName)) || string.IsNullOrEmpty(tenantId))
```

Comment:
We should require subscription ID - we need it for all of the subscription calls

Reply:
I cannot separate the AccessToken and SkipValidation parameter sets because there is no mandatory parameter in AccessToken that is not mandatory in SkipValidation (I end up getting parameter set cannot be resolved when trying to call the AccessToken parameter set).  That is why SubscriptionID can't be mandatory so I need this check.